src/coreclr/jit/targetriscv64.cpp

```diff
@@ -24,4 +24,154 @@ const regNumber fltArgRegs [] = {REG_FLTARG_0, REG_FLTARG_1, REG_FLTARG_2, REG_F
+        assert(size > 0);
+        assert(size <= TARGET_POINTER_SIZE);
+        if (inFloatReg)
+        {
+            return ABIPassingSegment::InRegister(m_floatRegs.Dequeue(), offset, size);
+        }
+        else if (m_intRegs.Count() > 0)
+        {
+            return ABIPassingSegment::InRegister(m_intRegs.Dequeue(), offset, size);
+        }
+        else
+        {
+            assert((m_stackArgSize % TARGET_POINTER_SIZE) == 0);
+            ABIPassingSegment seg = ABIPassingSegment::OnStack(m_stackArgSize, offset, size);
+            m_stackArgSize += TARGET_POINTER_SIZE;
+            return seg;
+        }
+    };
+
+    if ((floatFields > 0) && (m_floatRegs.Count() >= floatFields) && (m_intRegs.Count() >= intFields))
+    {
+        // Hardware floating-point calling convention
+        if ((floatFields == 1) && (intFields == 0))
+        {
+            if (flags == STRUCT_NO_FLOAT_FIELD)
+                assert(varTypeIsFloating(type)); // standalone floating-point real
+            else
+                assert((flags & STRUCT_FLOAT_FIELD_ONLY_ONE) != 0); // struct containing just one FP real
+
+            return ABIPassingInformation::FromSegment(comp, ABIPassingSegment::InRegister(m_floatRegs.Dequeue(), 0,
+                                                                                          passedSize));
+        }
+        else
+        {
+            assert(varTypeIsStruct(type));
+            assert((floatFields + intFields) == 2);
+            assert(flags != STRUCT_NO_FLOAT_FIELD);
+            assert((flags & STRUCT_FLOAT_FIELD_ONLY_ONE) == 0);
+
+            unsigned firstSize  = ((flags & STRUCT_FIRST_FIELD_SIZE_IS8) != 0) ? 8 : 4;
+            unsigned secondSize = ((flags & STRUCT_SECOND_FIELD_SIZE_IS8) != 0) ? 8 : 4;
+            unsigned offset = max(firstSize, secondSize); // TODO: cover empty fields and custom offsets / alignments
+
+            bool isFirstFloat  = (flags & (STRUCT_FLOAT_FIELD_ONLY_TWO | STRUCT_FLOAT_FIELD_FIRST)) != 0;
+            bool isSecondFloat = (flags & (STRUCT_FLOAT_FIELD_ONLY_TWO | STRUCT_FLOAT_FIELD_SECOND)) != 0;
+            assert(isFirstFloat || isSecondFloat);
+
+            return {2, new (comp, CMK_ABI) ABIPassingSegment[]{PassSlot(isFirstFloat, 0, firstSize),
+                                                               PassSlot(isSecondFloat, offset, secondSize)}};
+        }
+    }
+    else
+    {
+        // Integer calling convention
+        if (passedSize <= TARGET_POINTER_SIZE)
+        {
+            return ABIPassingInformation::FromSegment(comp, PassSlot(false, 0, passedSize));
+        }
+        else
+        {
+            assert(varTypeIsStruct(type));
+            return {2, new (comp, CMK_ABI)
+                           ABIPassingSegment[]{PassSlot(false, 0, TARGET_POINTER_SIZE),
+                                               PassSlot(false, TARGET_POINTER_SIZE, passedSize - TARGET_POINTER_SIZE)}};
+        }
+    }
+
+    unreached();
```

Comment:
Does this do anything? Won't we have a compiler error if this becomes accidentally reachable?

Feel free to remove this separately.